cgotorch/memory.cc

```diff
@@ -0,0 +1,13 @@
+// Copyright 2020, GoTorch Authors
+#include "torch/torch.h"
+
+// FIXME(shendiaomo): including cgotorch.h before torch/torch.h will fail
+#include "cgotorch/cgotorch.h"
+
+thread_local bool gcPrepared;
+
+uint8_t GCPrepared() { return gcPrepared; }
+
+void PrepareGC() { gcPrepared = true; }
```

Comment:
Do we need to make sure that the assignment is atomic?

Reply:
No, we don't. `gcPrepared` is a `thread_local`, each thread has its only copy.